modules/custom/az_migration/src/Plugin/migrate/source/AZNodeWithFieldCollection.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Drupal\az_migration\Plugin\migrate\source;
+
+use Drupal\migrate\Row;
+use Drupal\node\Plugin\migrate\source\d7\Node as D7Node;
+
+/**
+ * Extends D7Node source plugin to convert field collection content into paragraph content.
+ *
+ * Available configuration keys:
+ * - node_type: The node_types to get from the source - can be a string or
+ *   an array. If not declared then nodes of all types will be retrieved.
+ * - field_collection_names: The field_collection types to get from the source - can be a string or
+ *   an array. If not declared then nothing will be retrieved.
+ *
+ * Examples:
+ *
+ * @code
+ * source:
+ *   plugin: az_node_with_field_collection
+ *   node_type: page
+ *   field_collection_names: field_accordion
+ * @endcode
+ *
+ * In this example nodes of type page, and the accordion field collection are retrieved from the source database
+ *
+ * @code
+ * source:
+ *   plugin: az_node_with_field_collection
+ *   node_type: [page, test]
+ *   field_collection_names:
+ * .    - field_accordion
```

Comment:
Nitpick stray character.
```suggestion
 *     - field_accordion
```